src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java

```diff
@@ -1433,6 +1534,165 @@ private static int validateLogicalDecodingMessageExcludeList(Configuration confi
+                    
+                    try (Connection connection = ybConnection.connection();
+                          final Statement statement = connection.createStatement()) {
+                        statement.execute(createNeeded ? createQuery : alterQuery);
+                    } 
+                    return tableList;
+                } else {
+                    throw new DebeziumException("Invalid Autocreate Mode");
+                }
+            } catch (SQLException e) {
+                retryCount++;
+                exception = e;
+                if (retryCount > config.getInteger(MAX_CONNECTOR_RETRIES)) {
+                    LOGGER.error("Failed to extract table list from publication. All {} retries failed",config.getInteger(MAX_CONNECTOR_RETRIES), e);
+                    throw new DebeziumException(e);
+                }
+                LOGGER.warn("Error while trying to extract table list from publication. Will retry, attempt {} out of {}",
+                            retryCount, config.getInteger(MAX_CONNECTOR_RETRIES));
+                e.printStackTrace();
+
+                try {
+                    final Metronome retryMetronome = Metronome.parker(Duration.ofMillis(config.getLong(CONNECTOR_RETRY_DELAY_MS)), Clock.SYSTEM);
+                    retryMetronome.pause();
+                } catch (InterruptedException ie) {
+                    LOGGER.warn("Connector retry sleep interrupted by exception: {}", ie);
+                    Thread.currentThread().interrupt();
+                }
+            } catch (DebeziumException e) {
+                exception = e;
+                throw e;
+            }
+        }
+
+        if (exception != null) {
+            throw new DebeziumException(exception);
+        }
+
+        return null;
+    }
+
+    public static JdbcConfiguration getJdbcConfig(Configuration config) {
+            return JdbcConfiguration.copy(Configuration.empty())
+                    .with(JdbcConfiguration.DATABASE, config.getString(DATABASE_NAME))
+                    .with(JdbcConfiguration.HOSTNAME, config.getString(HOSTNAME))
+                    .with(JdbcConfiguration.PORT, config.getString(PORT))
+                    .with(JdbcConfiguration.USER, config.getString(USER))
+                    .with(JdbcConfiguration.PASSWORD, config.getString(PASSWORD))
+                    .withDefault(YugabyteDBConnectorConfig.MAX_RETRIES, config.getInteger(MAX_RETRIES))
+                    .with(YugabyteDBConnectorConfig.RETRY_DELAY_MS, config.getLong(RETRY_DELAY_MS))
+                    .build();
+    }
+
+    public static boolean checkIfPublicationExists(String publicationName, YugabyteDBConnection ybConnection) throws SQLException {
+        try (Connection connection = ybConnection.connection();
+              final Statement statement = connection.createStatement()) {
+            String query = "SELECT * FROM pg_publication_tables WHERE pubname = '" + publicationName + "';";
```

Comment:
`pg_publication_tables` lists all the tables that are part of a publication. There can be quite a few of them and we don't need to know them here.

Use `pg_publication` table here.
`SELECT * FROM pg_publication WHERE pubname = '__name__';`